diff --git a/mindspeed/te/pytorch/module/grouped_linear.py b/mindspeed/te/pytorch/module/grouped_linear.py
index ea164a81..08de7a59 100644
--- a/mindspeed/te/pytorch/module/grouped_linear.py
+++ b/mindspeed/te/pytorch/module/grouped_linear.py
@@ -33,7 +33,7 @@ class MindSpeedTEGroupedLinearGMM(torch.autograd.Function):
                 group_list_type=None,
                 ori_weight=None,
                 *weight_input_T) -> torch.Tensor:
-        
+
         # Due to ascend gmm kernal k split limitations, we need a tensor m_split, not a tensor List.
         # Also can be solved in token_dispatcher.
         if m_split is not torch.Tensor:
@@ -47,7 +47,7 @@ class MindSpeedTEGroupedLinearGMM(torch.autograd.Function):
         ctx.save_for_backward(input_tensor, *ori_weight)
         return fwd_output
 
-    @staticmethod  
+    @staticmethod
     def backward(ctx, grad_output):
 
         group_list = ctx.group_list
@@ -59,7 +59,7 @@ class MindSpeedTEGroupedLinearGMM(torch.autograd.Function):
         # K spilt gmm.
         grad_weight = torch_npu.npu_grouped_matmul([inp.T], [grad_output], bias=None, group_list=group_list,
                                     split_item=3, group_type=2, group_list_type=group_list_type)[0]
-        
+
         return grad, None, None, None, *grad_weight
 
 
@@ -126,15 +126,12 @@ class MindSpeedTEGroupedLinear(torch.nn.Module):
             setattr(param, 'allreduce', not (is_expert and self.expert_parallel))
 
     def forward(self, x, m_splits):
-        group_list_type = 1
-        for w in self.total_weight:
-            if self.parallel_mode == 'column':
-                w = w.view(self.config.hidden_size, -1)
-            else:
-                w = w.view(-1, self.config.hidden_size)
-        self.total_weight_T = [w.T for w in self.total_weight]
-        output = MindSpeedTEGroupedLinearGMM.apply(x, m_splits, group_list_type, self.total_weight, *self.total_weight_T)
-        return output, None
+        original_weight = torch.cat([w.t() for w in self.total_weight], dim=0)
+        if self.parallel_mode == "column":
+            w = original_weight.view(self.num_gemms, self.config.hidden_size, -1)
+        else:
+            w = original_weight.view(self.num_gemms, -1, self.config.hidden_size)
+        return Ops.gmm(x, w, torch.Tensor(m_splits).long(), trans_b=False, gemm_fusion=False, original_weight=original_weight), None
 
     def _sharded_state_dict_grouped(
             self, tp_axis_map, prefix='', sharded_offsets=(), metadata=None
@@ -299,7 +296,7 @@ class MindSpeedTERowParallelGroupedLinear(MindSpeedTEGroupedLinear):
         return super()._sharded_state_dict_grouped(
             tp_axis_map, prefix, sharded_offsets, metadata
         )
-    
+
 
 def mindspeed_groupedmlp_weighted_bias_swiglu_impl(x, bias, probs, fp8_input_store=False):
     """Patch of TEGroupedMLP with MindSpeed.
